be/src/service/service_be/starrocks_be.cpp

```diff
@@ -200,6 +200,11 @@ void start_be(const std::vector<StorePath>& paths, bool as_cn) {
         LOG(INFO) << process_name << " starts by skipping the datacache initialization";
     }
 
+    // set up thrift client before providing any service to the external
+    // because these services may use thrift client, for example, stream
+    // load will send thrift rpc to FE after http server is started
+    ThriftRpcHelper::setup(exec_env);
```

Comment:
FE will only redirect the stream load request to the alive backends. If the heartbeat server is setup, then FE will redirect the request and it will cause failure because of HTTP is not ready.
If you start thrift before HTTP, we need to return OK for all heartbeat requests after all server is ready.

Reply:
1. The initial state of BE in FE is restored from checkpoint, and if the BE was alive before stopping the FE, the FE will restore the state as alive. So it's possible FE will redirect the stream load to BE regardless of the newest state of the BE when FE is just started. This is the case of the problem.
2. We don't adjust the order to start thrift server. `ThriftRpcHelper` plays the role as a thrift client which is different from thrift server

Reply:
wondering if we can move all the service start as near as possible to minimize the window of service unavailable.